pennylane_qiskit/qiskit_device.py

```diff
@@ -180,61 +166,122 @@ def __init__(self, wires, provider, backend, shots=1024, **kwargs):
 
-    def apply(self, operation, wires, par):
-        mapped_operation = self._operation_map[operation]
+    def reset(self):
+        self._reg = QuantumRegister(self.num_wires, "q")
+        self._creg = ClassicalRegister(self.num_wires, "c")
+        self._circuit = QuantumCircuit(self._reg, self._creg, name="temp")
 
-        qregs = [self._reg[i] for i in wires]
+        self._current_job = None
+        self._state = None  # statevector of a simulator backend
 
-        if operation == "QubitStateVector":
+    def apply(self, operations, **kwargs):
+        rotations = kwargs.get("rotations", [])
 
-            if self.backend_name == "unitary_simulator":
-                raise QuantumFunctionError(
-                    "The QubitStateVector operation is not supported on the unitary simulator backend."
-                )
+        applied_operations = self.apply_operations(operations)
+        rotation_circuits = self.apply_operations(rotations)
+        applied_operations.extend(rotation_circuits)
 
-            if len(par[0]) != 2 ** len(wires):
-                raise ValueError("State vector must be of length 2**wires.")
+        for circuit in applied_operations:
+            self._circuit += circuit
 
-            qregs = list(reversed(qregs))
+        if self.backend_name not in self._state_backends:
+            # Add measurements if they are needed
+            for qr, cr in zip(self._reg, self._creg):
+                measure(self._circuit, qr, cr)
 
-            # TODO: Once a fix is available in Qiskit-Aer, remove the following:
-            par = (x.tolist() for x in par if isinstance(x, np.ndarray))
+        # These operations need to run for all devices
+        qobj = self.compile()
+        self.run(qobj)
 
-        if operation == "QubitUnitary":
+    def apply_operations(self, operations):
+        """Apply the circuit operations.
 
-            if len(par[0]) != 2 ** len(wires):
-                raise ValueError("Unitary matrix must be of shape (2**wires, 2**wires).")
+        This method serves as an auxiliary method to :meth:`~.QiskitDevice.apply`.
+
+        Args:
+            operations (List[pennylane.Operation]): operations that rotate into the
+                measurement basis
+
+        Returns:
+            list[QuantumCircuit]: a list of quantum circuit objects that
+                specify the corresponding operations
+        """
+        circuits = []
 
-            qregs = list(reversed(qregs))
+        for operation in operations:
+            # Apply the circuit operations
+            wires = operation.wires
+            par = operation.parameters
+            operation = operation.name
 
-        dag = circuit_to_dag(QuantumCircuit(self._reg, self._creg, name=""))
-        gate = mapped_operation(*par)
+            mapped_operation = self._operation_map[operation]
+
+            # TODO: Once a fix is available in Qiskit-Aer, remove the
+            # following:
```

Comment:
Is there a github issue related to this? Might be good to provide a link to it in the comment!

Reply:
Oh right, sorry this was coming from this recent PR:
https://github.com/XanaduAI/pennylane-qiskit/pull/81

And the specific issue is:
https://github.com/Qiskit/qiskit-aer/issues/692

Although the issue was fixed, it's not in the pip version yet (neither `0.18.0` nor `0.18.1`). Guess we'll have to wait until `0.19.0`.

Reply:
Updated the comment such that it refers to the release. Also set on watching the `Qiskit-Aer` repo for new releases :wink: